src/main/java/org/myexample/KittenStatistics.java

```diff
@@ -19,76 +20,160 @@ public void setKittens(List<Kitten> kittens) {
     /**
      * Находит средний возраст котят в списке.
      * Используйте обычные циклы итд
+     *
      * @return Средний возраст котят в виде double.
      */
     public double findKittensAverageAge() {
-        return 0;
+        int qty = kittens.size();
+        int sumOfages = 0;
+        for (Kitten kitten : kittens) {
+            sumOfages = sumOfages + kitten.getAge();
+        }
+        return (double) sumOfages / qty;
     }
 
     /**
      * Находит самого старшего котенка в списке.
      * Используйте обычные циклы итд
+     *
      * @return Самый старший котенок в виде Optional.
      */
     public Optional<Kitten> findOldestKitten() {
-        return null;
+        int max = 0;
+        Optional<Kitten> oldestKitten = Optional.empty();
+        for (Kitten value : kittens) {
+            int age = value.getAge();
+            if (age > max) {
+                max = age;
+            }
+        }
```

Comment:
You can already save the current oldest kitten in a variable `Kitten oldest = null;` and then return `Optional.ofNullable(oldest);`